todo_list_app/views.py

```diff
@@ -25,32 +25,43 @@ def post(self, request):
         serializer = self.serializer_class(data=request.data)
         if serializer.is_valid(raise_exception=True):
             serializer.save()
-            user = serializer.data
-            # send_code_to_user(user['email'])
-            return Response({'data': user, 'message': f"hi thanks for signing up passcode"},
+            user_data = serializer.data
+            user = User.objects.get(email=user_data['email'])
+            send_code_to_user(user_data['email'], user.id)
+            user_id = urlsafe_base64_encode(smart_bytes(user.id))
+            return Response({'data': user_data, 'user_id': user_id, 'message': f"hi thanks for signing up passcode"},
                             status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-class LoginError(Exception):
-    pass
-
+class VerifyUserEmailView(GenericAPIView):
 
-def verify_user_email(request, user_id):
-    if request.method == "POST":
-        otp_code = request.POST['code']
+    def post(self, request):
+        otp_code = request.data.get('code')
+        user_id = request.data.get('user_id')
+        user_id = int(force_str(urlsafe_base64_decode(user_id)))
         try:
-            user_id = int(force_str(urlsafe_base64_decode(user_id)))
+            if not verify_otp_code(otp_code, user_id):
+                return Response({
+                    'message': 'code is Invalid'
```

Comment:
Every message returned from an API should have proper grammar and correct upper/lower case characters. For this case, the message should be either:
```suggestion
                    'message': 'Code is Invalid'
```
OR
```suggestion
                    'message': 'Invalid Code'
```

Reply:
Update every message if it's not correct.